### PR TITLE
Produce TinyV2 mappings

### PR DESCRIPTION
**Pull request by natanfudge**

### CODE REVIEW

**Comment by modmuss50** [comment]
I think this is the wrong way to go about this.
Gradle should generate the v2 mappings from all the v1 mappings and publish them to maven in a jar file to be used where needed. There is no point having duplicate data on an already large repo.

**Comment by natanfudge** [comment]
OK, I'll do that.

**Review by Player3324 — APPROVED** [review]

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
+/workspace.xml
```
Comment (modmuss50) [inline .idea/.gitignore]: This can go in the root .gitignore? any reason for this?

File: build.gradle [hunk 2]
```diff
@@ -1,39 +1,79 @@
...
+    repositories{
+		maven {
+			name "Fabric Repository"
+			url 'https://maven.fabricmc.net'
+		}
+		mavenLocal()
+    }
+
+    dependencies {
+		classpath "cuchaz:enigma:0.14.1-local"
```
Comment (modmuss50) [inline build.gradle]: Cannot merge untill these are the versions on git, leaving this here as a note.

File: build.gradle [hunk 3]
```diff
@@ -1,39 +1,79 @@
...
+		classpath "net.fabricmc:stitch:0.2.1-local"
+	}
+}
+
 plugins {
     id 'maven'
     id 'maven-publish'
 }
 
+mkdir("localMappings")
```
Comment (modmuss50) [inline build.gradle]: Move this into the build dir

File: build.gradle [hunk 4]
```diff
@@ -1,39 +1,79 @@
...
+
 file('mappings').eachFile {
-    if (!it.name.contains(".tiny")) return
+    if(!it.name.endsWith(".tiny")) return
+    File v1MappingFile = it
+    File v2MappingFile = new File(System.getProperty("user.dir") + "/localMappings/${it.name}")
+    if(!v2MappingFile.exists()){
```
Comment (modmuss50) [inline build.gradle]: This should become a task with propper inputs and outputs I think, would ensure the caching is done right, as right now if the v1 changes, the v2 wont be updated.

**Review by modmuss50 — CHANGES_REQUESTED** [review]
This needs some work to ensure that changes are handled correctally.

**Comment by modmuss50** [comment]
If you need help with some of the gradle stuff here let me know.

File: build.gradle [hunk 5]
```diff
@@ -1,39 +1,79 @@
...
+		maven {
+			name "Fabric Repository"
+			url 'https://maven.fabricmc.net'
+		}
+		mavenLocal()
+    }
+
+    dependencies {
+		classpath "cuchaz:enigma:0.14.1-local"
+		classpath "net.fabricmc:stitch:0.2.1-local"
```
Comment (modmuss50) [inline build.gradle]: What is this used for? cannot see any inports for it?

**Review by modmuss50 — CHANGES_REQUESTED** [review]

File: .idea/.gitignore (hunk 1 above)
Comment (natanfudge) [reply .idea/.gitignore]: Don't know how it got here

Comment (natanfudge) [reply .idea/.gitignore]: Ah, the /.idea directory should not be included at all

File: build.gradle (hunk 5 above)
Comment (natanfudge) [reply build.gradle]: removed

File: build.gradle (hunk 4 above)
Comment (natanfudge) [reply build.gradle]: Made it so conversion is done on demand (and is cached properly)

**Review by modmuss50 — CHANGES_REQUESTED** [review]
Why delete the 1.14.1.tiny file? Please add back.

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,4 @@
+localMappings/
```
Comment (modmuss50) [inline .gitignore]: This line here shoudlnt be needed now.

File: build.gradle [hunk 7]
```diff
@@ -1,50 +1,100 @@
...
-			url "http://mavenupload.modmuss50.me/"
-			if (project.hasProperty('mavenPass')) {
-				credentials {
-					username 'buildslave'
-					password project.getProperty('mavenPass')
-				}
-			}
-		}
-	}
+    repositories {
```
Comment (modmuss50) [inline build.gradle]: formatting changes here, not needed

File: build.gradle [hunk 8]
```diff
@@ -1,50 +1,100 @@
...
+    }
+}
+
 plugins {
     id 'maven'
     id 'maven-publish'
 }
 
+def localMappingsPath = "$buildDir/v2Mappings"
+mkdir(localMappingsPath)
```
Comment (liach) [inline build.gradle]:
```suggestion
localMappingsPath.mkdirs()
```

Comment (natanfudge) [reply build.gradle]: seriously?

Comment (liach) [reply build.gradle]: 🤔 what

Comment (natanfudge) [reply build.gradle]: Oh, I missed the extra `s` at the end of `mkdirs`. My bad

Comment (liach) [reply build.gradle]: I realized my fault and added `s` :sweat:

Comment (natanfudge) [reply build.gradle]: image

File: build.gradle [hunk 9]
```diff
@@ -1,37 +1,87 @@
...
+
+	dependencies {
+		classpath "cuchaz:enigma:0.15.0"
+	}
+}
+
 plugins {
-    id 'maven'
-    id 'maven-publish'
+	id 'maven'
```
Comment (natanfudge) [inline build.gradle:17]: What can I do... The indent wasn't consistent...

Comment (liach) [reply build.gradle:17]: Always use tabs.

**Review by liach — APPROVED** [review]

**Comment by modmuss50** [comment]
This seems good. Needs engima first though.

**Comment by natanfudge** [comment]
In Enigma I also need @liach to fix it to not produce blank mappings.

**Comment by natanfudge** [comment]
This has been resolved.

**Review by Player3324 — APPROVED** [review]

File: build.gradle [hunk 10]
```diff
@@ -1,39 +1,87 @@
...
+					classifier = "v2"
+				}
+			}
+		}
+	}
+
 }
 
+def makeJar(String mcVersion, File mappings, boolean v2) {
+	def jarFilename = "intermediary-" + mcVersion + (v2 ? "v2" : "")
```
Comment (modmuss50) [inline build.gradle]: For the filename make it `-v2`.

**Review by modmuss50 — CHANGES_REQUESTED** [review]
One minor thing. After this it looks good.

Tested and seems to work well.

File: build.gradle (hunk 10 above)
Comment (natanfudge) [reply build.gradle]: can you please do the "suggest changes" so I can auto commit

Comment (modmuss50) [reply build.gradle]: Yeah, let me figure out how to do that :D

Comment (modmuss50) [inline build.gradle]:
```suggestion
	def jarFilename = "intermediary-" + mcVersion + (v2 ? "-v2" : "")
```

Comment (natanfudge) [reply build.gradle]: thanks

**Review by liach — APPROVED** [review]

**Pull request merged**